### ISSUE TITLE
Regularization cost equals 0?

### ISSUE THREAD

**Issue by RyanTang1** [wenwei202/iss-rnns#8]
Hello:
  As I'm trying to understand your code, using this commands

python ptb_word_lm.py --model sparselarge --data_path simple-examples/data/  --config_file structure_grouplasso.json

I found that when I'm trying to print out your regularization cost in run epochs function, I keep
getting 0. Is there something I missed? Thank you.

**Comment by wenwei202** [comment]
How do you change your code? There is no sufficient info for me to identify the problem.

**Comment by RyanTang1** [comment]
Hello:
    Here is what I added:

    vals = session.run(fetches, feed_dict)
    cost = vals["cost"]
    state = vals["final_state"]
    print(vals["regularization"])
    costs += cost
    regularizations += vals["regularization"]

I simply added a print (vals["regularization"]) in the middle. I didn't change other stuffs, thanks for helping!

**Comment by RyanTang1** [comment]
Hello:
  I just want to check if something went wrong. Sorry to bother you again but do you have any idea what's going on? Also, below is a epoch of the code using the command I mentioned. The regularization are all zeros. I'm not sure if I do something wrong.


Epoch: 1 Learning rate: 1.000
0.004 perplexity: 5153.357  cost: 8.5474  regularization: 0.0000  total_cost: 8.5474   speed: 838 wps
0.104 perplexity: 848.609  cost: 6.7436  regularization: 0.0000  total_cost: 6.7436   speed: 1054 wps
0.204 perplexity: 650.397  cost: 6.4776  regularization: 0.0000  total_cost: 6.4776   speed: 1475 wps
0.304 perplexity: 541.483  cost: 6.2943  regularization: 0.0000  total_cost: 6.2943   speed: 1738 wps
0.404 perplexity: 477.586  cost: 6.1687  regularization: 0.0000  total_cost: 6.1687   speed: 1912 wps
0.504 perplexity: 436.181  cost: 6.0781  regularization: 0.0000  total_cost: 6.0781   speed: 2031 wps
0.604 perplexity: 399.405  cost: 5.9900  regularization: 0.0000  total_cost: 5.9900   speed: 2109 wps
0.703 perplexity: 373.979  cost: 5.9242  regularization: 0.0000  total_cost: 5.9242   speed: 2167 wps
0.803 perplexity: 353.959  cost: 5.8692  regularization: 0.0000  total_cost: 5.8692   speed: 2211 wps
0.903 perplexity: 335.614  cost: 5.8160  regularization: 0.0000  total_cost: 5.8160   speed: 2259 wps
Epoch: 1 Train Perplexity: 321.701   regularization: 0.0000

**Comment by RyanTang1** [comment]
Hello:
  I think I found the problem. Apparently, for some reasons, right now the the operation name has changed from  Model/RNN/multi_rnn_cell/cell_0/basic_lstm_cell/weights to Model/RNN/multi_rnn_cell/cell_0/basic_lstm_cell/kernel. I found this after I print the operation name out. Such a hard bug to find. Anyway, thanks for the help!
